src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -477,8 +477,8 @@ public override int Execute(DocumentsOperationContext context)
                                         var item = Database.DocumentsStorage.GetDocumentOrTombstone(context, cmd.Id);
                                         if (item.Missing)
                                         {
-                                            var missingMsg = $"This shouldn't happened and probably a bug! Missing id: {cmd.Id}, doc: {_count}, raft:{current} on {context.DocumentDatabase.ServerStore.NodeTag}";
-                                            throw new InvalidDataException(missingMsg);
+                                            // this item is missing and there is nothing we can do about it.
+                                            continue;
```

Comment:
This is going to cause issues down the line. If the value is missing, you are skipping it, and in that case, the result to the client will have one item less, no?

Reply:
Yes, the only valid scenario that I can think, is if the deleted document was purged, right? 
So, should we recreate the document in that case?